Bukkit/0087-Add-PlayerSettingsUpdateEvent.patch

```diff
@@ -0,0 +1,389 @@
++    }
++
++    public String getNewLocale() {
++        return getNewUpdate().getLocale();
++    }
++
++    public String getOldLocale() {
++        return hasUpdated() ? getOldUpdate().getLocale() : null;
++    }
++
++    public boolean didLocaleChange() {
++        return !new EqualsBuilder().append(getOldLocale(), getNewLocale()).isEquals();
++    }
++
++    public MainHand getNewMainHand() {
++        return getNewUpdate().getMainHand();
++    }
++
++    public MainHand getOldMainHand() {
++        return hasUpdated() ? getOldUpdate().getMainHand() : null;
++    }
++
++    public boolean didMainHandChange() {
++        return !new EqualsBuilder().append(getOldMainHand(), getNewMainHand()).isEquals();
++    }
++
++    public Integer getNewRenderDistance() {
++        return getNewUpdate().getRenderDistance();
++    }
++
++    public Integer getOldRenderDistance() {
++        return hasUpdated() ? getOldUpdate().getRenderDistance() : null;
++    }
++
++    public boolean didRenderDistanceChange() {
++        return !new EqualsBuilder().append(getOldRenderDistance(), getNewRenderDistance()).isEquals();
++    }
++
++    public ChatVisibility getNewChatVisibility() {
++        return getNewUpdate().getChatVisibility();
++    }
++
++    public ChatVisibility getOldChatVisibility() {
++        return hasUpdated() ? getOldUpdate().getChatVisibility() : null;
++    }
++
++    public boolean didChatVisibilityChange() {
++        return !new EqualsBuilder().append(getOldChatVisibility(), getNewChatVisibility()).isEquals();
++    }
++
++    public Boolean getNewChatColors() {
++        return getNewUpdate().getChatColors();
++    }
++
++    public Boolean getOldChatColors() {
++        return hasUpdated() ? getOldUpdate().getChatColors() : null;
++    }
++
++    public boolean didChatColorsChange() {
++        return !new EqualsBuilder().append(getOldChatColors(), getNewChatColors()).isEquals();
++    }
++
++    public ImmutableSet<Skin.Part> getNewSkinParts() {
++        return getNewUpdate().getSkinParts();
++    }
++
++    public ImmutableSet<Skin.Part> getOldSkinParts() {
++        return hasUpdated() ? getOldUpdate().getSkinParts() : null;
++    }
++
++    public boolean didSkinPartsChange() {
++        return !new EqualsBuilder().append(getOldSkinParts(), getNewSkinParts()).isEquals();
++    }
++
++    @Override
++    public HandlerList getHandlers() {
++        return handlers;
++    }
++
++}
+diff --git a/src/main/java/org/bukkit/util/EnumUtils.java b/src/main/java/org/bukkit/util/EnumUtils.java
+index d064568..7cb4c29 100644
+--- a/src/main/java/org/bukkit/util/EnumUtils.java
++++ b/src/main/java/org/bukkit/util/EnumUtils.java
+@@ -19,4 +19,19 @@ public final class EnumUtils {
+         copy.addAll(set);
+         return copy;
+     }
++
++    /**
++     * Transform one enum to another enum type based on {@link Enum#ordinal()}.
++     */
++    public static <F extends Enum<F>, T extends Enum<T>> T transform(F from, Class<T> to, T def) {
```

Comment:
This should probably throw an exception if the enums have different lengths. I would also keep it in CraftBukkit, doesn't need to be in the public API.